host_tools/fw_update.py

```diff
@@ -80,18 +102,26 @@ def scan_for_device(self):
         ports = [x[0] for x in list_ports.comports()]
         retport = None
         self._log("Searching for an active bootloader")
-        
+
         for p in ports:
             try:
                 rsp = self._send_ping(p)
                 if not rsp == None:
-                    self._log("Found bootloader on: {}".format(p))
+                    ver = self._get_version(p)
+                    if not ver:
+                        version = '<=0.1.2'
+                    else:
+                        version = '.'.join(
+                            [str(x) for x in
+                             (ver.major, ver.minor, ver.bugfix)])
+                    self._log("Found bootloader ver: {} on: {}".format(version, p))
                     retport = p
                     break
                 else:
                     raise Exception()
 
-            except:
+            except Exception as e:
```

Comment:
Well this is not good.  Is there a more explicit exception to catch here or series of Exceptions.  I'm quite sure you don't want to catch a SIGINT here in example.